plugins/modules/inventory_intent.py

```diff
@@ -1992,13 +2033,114 @@ def get_device_ids(self, device_ips):
+        """
+
+        device_ips = []
+        for hostname in hostname_list:
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"hostname": hostname}
+                )
+                if response:
+                    self.log("Received API response from 'get_device_list': {0}".format(str(response)), "DEBUG")
+                    response = response.get("response")
+                    if not response:
+                        continue
+                    device_ip = response[0]["managementIpAddress"]
+                    device_ips.append(device_ip)
+            except Exception as e:
+                error_message = "Error while fetching device from Cisco Catalyst Center - {0}".format(str(e))
+                self.log(error_message, "ERROR")
+
+        return device_ips
+
+    def get_device_ips_from_serial_number(self, serial_number_list):
+        """
+        Get the list of unique device IDs for list of specified serial number of devices in Cisco Catalyst Center.
+        Parameters:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            serial_number_list (list): The list of serial number of devices for which you want to retrieve the device IDs.
+        Returns:
+            list: The list of unique device IDs for the specified devices with serial number.
+        Description:
+            Queries Cisco Catalyst Center to retrieve the unique device ID associated with a device having the specified
+            serial number.If the device is not found in Cisco Catalyst Center, then print the log message with error severity.
+        """
+
+        device_ips = []
+        for serial_number in serial_number_list:
+            try:
+                response = self.dnac._exec(
+                    family="devices",
+                    function='get_device_list',
+                    params={"serialNumber": serial_number}
+                )
+                if response:
+                    self.log("Received API response from 'get_device_list': {0}".format(str(response)), "DEBUG")
+                    response = response.get("response")
+                    if not response:
+                        continue
+                    device_ip = response[0]["managementIpAddress"]
+                    device_ips.append(device_ip)
+            except Exception as e:
+                error_message = "Error while fetching device from Cisco Catalyst Center - {0}".format(str(e))
+                self.log(error_message, "ERROR")
+
+        return device_ips
+
+    def get_device_ips_from_mac_address(self, mac_address_list):
+        """
+        Get the list of unique device IDs for list of specified mac address of devices in Cisco Catalyst Center.
```

Comment:
same comments applicable this API as well..

Reply:
added.